### PR TITLE
feat: implements DIM STS adapter

### LINKED ISSUE #1038: STS: DIM integration
## WHAT

Provide a `SecureTokenService` implementation that integrates with DIM

### PR DESCRIPTION
**Pull request by wolf4ood**
## WHAT

Implements a `SecureTokenService` that talks with DIM.

For obtaining an SI token with this implementation first we need to fetch a valid OAuth token with the `DimOauth2Client`, then we make two kind of request to DIM

- `grantAccess` for fetching an SI with the `bearerAccessScope` 
- `signToken`  for fetching an SI token providing the extracted `access_token` from the DSP request (SI token)

The implementation of `DimOauth2Client` is currently sub-optimal since it's not caching the OAuth token. This will be implemented in subsequent PRs

This PR also ships an E2E test mocking the `OAuth` server and the DIM interaction

## WHY

iatp integration

Closes #1038

### CODE REVIEW

File: edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimSecureTokenService.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static org.eclipse.edc.identitytrust.SelfIssuedTokenConstants.PRESENTATION_ACCESS_TOKEN_CLAIM;
+import static org.eclipse.edc.jwt.spi.JwtRegisteredClaimNames.AUDIENCE;
+import static org.eclipse.edc.jwt.spi.JwtRegisteredClaimNames.ISSUER;
+import static org.eclipse.edc.jwt.spi.JwtRegisteredClaimNames.SUBJECT;
+import static org.eclipse.edc.spi.http.FallbackFactories.retryWhenStatusIsNotIn;
+
+public class DimSecureTokenService implements SecureTokenService {
```
Comment (paullatzelsperger) [inline edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimSecureTokenService.java:56]: doc?

**Review by paullatzelsperger — APPROVED** [review]

File: edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimOauthClientExtension.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.tractusx.edc.iam.iatp.sts.dim.oauth.DimOauth2Client;
+import org.eclipse.tractusx.edc.iam.iatp.sts.dim.oauth.DimOauthClientImpl;
+
+@Extension(DimOauthClientExtension.NAME)
+public class DimOauthClientExtension implements ServiceExtension {
```
Comment (jimmarino) [inline edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimOauthClientExtension.java:32]: Doc

File: edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimSecureTokenService.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    public static final MediaType TYPE_JSON = MediaType.parse("application/json");
+    public static final String READ_SCOPE = "read";
+    public static final String GRANT_ACCESS = "grantAccess";
+    public static final String SCOPE = "scope";
+    public static final String CREDENTIAL_TYPES = "credentialTypes";
+    public static final String SIGN_TOKEN = "signToken";
+
+    private final EdcHttpClient httpClient;
```
Comment (jimmarino) [inline edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimSecureTokenService.java:65]: Do we need these finals?

Comment (jimmarino) [inline edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/DimSecureTokenService.java:65]: This class would be a good candidate to refactor in a subsequent PR. It's starting to get a bit large

File: edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/StsRemoteClientConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.edc.iam.iatp.sts.dim;
+
+/**
+ * Configuration of the OAuth2 client
+ */
+public record StsRemoteClientConfiguration(String tokenUrl, String clientId, String clientSecretAlias) {
```
Comment (jimmarino) [inline edc-extensions/iatp/tx-iatp-sts-dim/src/main/java/org/eclipse/tractusx/edc/iam/iatp/sts/dim/StsRemoteClientConfiguration.java:25]: Doc

File: edc-extensions/iatp/tx-iatp-sts-dim/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ ********************************************************************************/
+
+plugins {
+    `java-library`
+    `maven-publish`
+}
+
+dependencies {
+    implementation(libs.edc.spi.core)
+    implementation(libs.edc.spi.policyengine)
```
Comment (jimmarino) [inline edc-extensions/iatp/tx-iatp-sts-dim/build.gradle.kts:27]: Are the above two depedencies required?

**Review by jimmarino — CHANGES_REQUESTED** [review]

File: edc-extensions/iatp/tx-iatp-sts-dim/build.gradle.kts (hunk 5 above)
Comment (wolf4ood) [reply edc-extensions/iatp/tx-iatp-sts-dim/build.gradle.kts]: removed

**Pull request merged**